### PR TITLE
MULE-13856: Begin a TX inside of another should make the app fail ins…

### PR DESCRIPTION
**Pull request by estebanwasinger**
…tead of commit the original TX

### CODE REVIEW

File: core-tests/src/test/java/org/mule/runtime/core/api/execution/TransactionalExecutionTemplateTestCase.java [hunk 1]
```diff
@@ -122,8 +128,8 @@ public void testActionNoneAndTxForCommit() throws Exception {
     MuleTransactionConfig config = new MuleTransactionConfig(TransactionConfig.ACTION_NONE);
     ExecutionTemplate executionTemplate = createExecutionTemplate(config);
     Object result = executionTemplate.execute(getEmptyTransactionCallback());
-    assertThat((CoreEvent) result, is(RETURN_VALUE));
-    verify(mockTransaction).commit();
+    assertThat(result, is(RETURN_VALUE));
+    //    verify(mockTransaction).commit();
```
Comment (marianogonzalez) [inline core-tests/src/test/java/org/mule/runtime/core/api/execution/TransactionalExecutionTemplateTestCase.java]: remove?

File: core-tests/src/test/java/org/mule/runtime/core/api/execution/TransactionalExecutionTemplateTestCase.java [hunk 2]
```diff
@@ -122,8 +128,8 @@ public void testActionNoneAndTxForCommit() throws Exception {
...
     verify(mockTransaction, VerificationModeFactory.times(0)).rollback();
```
Comment (marianogonzalez) [inline core-tests/src/test/java/org/mule/runtime/core/api/execution/TransactionalExecutionTemplateTestCase.java]: weird that you assert for rollback but the test name says commit

File: core-tests/src/test/java/org/mule/runtime/core/api/execution/TransactionalExecutionTemplateTestCase.java [hunk 3]
```diff
@@ -135,9 +141,9 @@ public void testActionNoneAndTxForRollback() throws Exception {
     MuleTransactionConfig config = new MuleTransactionConfig(TransactionConfig.ACTION_NONE);
     ExecutionTemplate executionTemplate = createExecutionTemplate(config);
     Object result = executionTemplate.execute(getEmptyTransactionCallback());
-    assertThat((CoreEvent) result, is(RETURN_VALUE));
-    verify(mockTransaction).rollback();
+    assertThat(result, is(RETURN_VALUE));
     verify(mockTransaction, VerificationModeFactory.times(0)).commit();
+    verify(mockTransaction, VerificationModeFactory.times(0)).rollback();
```
Comment (marianogonzalez) [inline core-tests/src/test/java/org/mule/runtime/core/api/execution/TransactionalExecutionTemplateTestCase.java]: commit AND rollback?

File: core-tests/src/test/java/org/mule/runtime/core/api/execution/TransactionalExecutionTemplateTestCase.java [hunk 4]
```diff
@@ -187,64 +193,47 @@ public void testActionNoneAndWithExternalTransactionWithTx() throws Exception {
...
-    MuleTransactionConfig config = new MuleTransactionConfig(TransactionConfig.ACTION_ALWAYS_BEGIN);
-    ExecutionTemplate executionTemplate = createExecutionTemplate(config);
-    config.setFactory(new TestTransactionFactory(mockTransaction));
-    Object result = executionTemplate.execute(getEmptyTransactionCallback());
-    assertThat((CoreEvent) result, is(RETURN_VALUE));
-    verify(mockTransaction).commit();
+    assertThat(result, is(RETURN_VALUE));
     verify(mockTransaction, VerificationModeFactory.times(0)).rollback();
-    assertThat(TransactionCoordination.getInstance().getTransaction(), IsNull.<Object>nullValue());
+    verify(mockTransaction, VerificationModeFactory.times(0)).commit();
```
Comment (marianogonzalez) [inline core-tests/src/test/java/org/mule/runtime/core/api/execution/TransactionalExecutionTemplateTestCase.java]: rollback and commit? Not only weird but, does order matter?

File: core-tests/src/test/java/org/mule/runtime/core/api/execution/TransactionalExecutionTemplateTestCase.java (hunk 1 above)
Comment (pablolagreca) [inline core-tests/src/test/java/org/mule/runtime/core/api/execution/TransactionalExecutionTemplateTestCase.java]: remove or fix

File: core/src/main/java/org/mule/runtime/core/internal/execution/ValidateTransactionalStateInterceptor.java [hunk 5]
```diff
@@ -23,13 +25,17 @@ public ValidateTransactionalStateInterceptor(ExecutionInterceptor<T> next, Trans
     this.transactionConfig = transactionConfig;
   }
 
+  //TODO ADD VALIDATION
```
Comment (pablolagreca) [inline core/src/main/java/org/mule/runtime/core/internal/execution/ValidateTransactionalStateInterceptor.java]: ???

**Comment by estebanwasinger** [comment]
@marianogonzalez 
* new transaction fails to start due to previous one ✅ 
* everything works as expected ✅  (Any other test do this)
* add a test of a transactional operation successfully joining an existing Tx (if we don't already have one) ✅

**Pull request merged**